app/uk/gov/hmrc/gform/formtemplate/BooleanExprValidator.scala

```diff
@@ -80,6 +81,25 @@ class BooleanExprValidator(
         .getOrElse(Invalid(s"page id '${fcId.value}' does not exist in the form"))
     }
 
+  private def validateIdList(fields: Seq[FormCtx]): List[ValidationResult] = {
+    val fcIds = fields.flatMap(extractFcIds)
+
+    fields
+      .map(f =>
+        addToListComponentIds
+          .find(_ === f.formComponentId)
+          .map { _ =>
+            Valid
+          }
+          .getOrElse(
+            Invalid(
+              s"id '${f.formComponentId}' named in $wrapperType expression does not exist in an Add To List component"
+            )
+          )
+      )
+      .toList ++ fcIds.map(validateIdIdx)
+  }
```

Comment:
Can we move this logic out of `BooleanExprValidator`? Despite its name this class should be used only for checking forward references.

Checking of invalid references is done in `validateInvalidReferences` of `FormTemplateValidator`

Reply:
Hi @josef-vlach, can you please review the updates I've made based on your review?